### DISCUSSION TITLE
Empty families for variants

### DISCUSSION THREAD

**Discussion by NLSVTN** [broadinstitute/seqr#2362] [Q&A] [answered]
We are a bit confused when in some cases we get no families for variants after the query is run. We went through the code and when **_named_family_sample_q** is not added into the final query:

https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/es_search.py#L1261

**matched_queries** will be absent:

```
if hasattr(raw_hit.meta, 'matched_queries'):
    family_guids = list(raw_hit.meta.matched_queries)
```

https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/es_search.py#L656-L657

So, the code in our case goes to the _3rd else_ statement:

```
family_guids = [family_guid for family_guid, samples_by_id in index_family_samples.items()
    if any(sample_id in alt_allele_samples and _is_matched_sample(family_guid, sample)
        for sample_id, sample in samples_by_id.items())]
```

https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/es_search.py#L675-L677

and there _sample_id in alt_allele_samples_ is always False, so **sample_ids** that were added at the very beginning when **EsSearch** was created (https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/es_search.py#L40-L41) are absent in **alt_allele_samples** retrieved from Elasticsearch.

So, now we are a bit confused as to how we could fix that. Is it so by design (then what are the reasons / theory?) or is it a bug in the program / data? Currently we see two solutions:

- Force the program to add named query which would make **matched_queries** non-empty or
- add somehow absent alt sample ids into postgres model: **self.samples_by_family_index[s.elasticsearch_index][s.individual.family.guid]**

If you could give any advice on that we would highly appreciate that.

**Comment by hanars** [comment, marked as answer]
seqr search has optimizations built in when you are searching across all samples in a project with either no inheritance filter or the "any affected" inheritance filter. The assumption is if you are searching across all samples in a project with no inheritance filter, any variant in the index is valid for at least one family. Similarly, if you apply the "any affected" filter it is faster to post process the results to map variants to families with affected individuals than to compute the expensive `matched_queries` operation.

The most common reason why you would be seeing variants with no families in this case would be if there are variants in your index not associated with any live samples in your project. This can happen if you do not apply an appropriate sample subset at the time of data loading, resulting in variants in your index not associated with any of the project families.
